packages/fxa-settings/src/pages/ResetPasswordRedesign/AccountRecoveryConfirmKeyOtp/container.tsx

```diff
@@ -0,0 +1,139 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import { RouteComponentProps, useLocation, useNavigate } from '@reach/router';
+import base32Decode from 'base32-decode';
+
+import { decryptRecoveryKeyData } from 'fxa-auth-client/lib/recoveryKey';
+import {
+  AuthUiErrors,
+  getLocalizedErrorMessage,
+} from '../../../lib/auth-errors/auth-errors';
+import { useAccount } from '../../../models';
+import { useFtlMsgResolver } from '../../../models/hooks';
+
+import {
+  AccountRecoveryConfirmKeyOtpContainerProps,
+  AccountRecoveryKeyOtpLocationState,
+} from './interfaces';
+
+import AccountRecoveryConfirmKeyOtp from '.';
+
+const AccountRecoveryConfirmKeyOtpContainer = ({
+  serviceName,
+}: AccountRecoveryConfirmKeyOtpContainerProps & RouteComponentProps) => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const account = useAccount();
+  const ftlMsgResolver = useFtlMsgResolver();
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const {
+    accountResetToken: previousAccountResetToken,
+    code,
+    email,
+    emailToHashWith,
+    estimatedSyncDeviceCount,
+    token,
+    uid,
+  } = location.state as AccountRecoveryKeyOtpLocationState;
+
+  // The password forgot code can only be used once to retrieve `accountResetToken`
+  // so we set its value after the first request for subsequent requests.
+  const [accountResetToken, setAccountResetToken] = useState(
+    previousAccountResetToken || ''
+  );
+
+  const getRecoveryBundleAndNavigate = async (
+    fetchedAccountResetToken: string,
+    recoveryKey: string
+  ) => {
+    try {
+      const { recoveryData, recoveryKeyId } =
+        await account.getRecoveryKeyBundle(
+          fetchedAccountResetToken,
+          recoveryKey,
+          uid
+        );
+
+      const decodedRecoveryKey = base32Decode(recoveryKey, 'Crockford');
+      const uint8RecoveryKey = new Uint8Array(decodedRecoveryKey);
+
+      const { kB } = await decryptRecoveryKeyData(
+        uint8RecoveryKey,
+        recoveryData,
+        uid
+      );
+
+      navigate(`/account_recovery_reset_password${window.location.search}`, {
+        state: {
+          accountResetToken: fetchedAccountResetToken,
+          email,
+          emailToHashWith,
+          estimatedSyncDeviceCount,
+          kB,
+          recoveryKeyId,
+        },
+      });
+    } catch (error) {
+      // TODO in FXA-7890 clearer error handling
```

Comment:
TODO in this PR?

Reply:
I've cleared all of the straggler TODOs marked with this issue number. Thank you!